### DISCUSSION TITLE
Deprecated field visibility

### DISCUSSION THREAD

**Discussion by pjungels-paraport** [graphql/graphiql#3191] [Q&A / Support]
Greetings!

I'm relatively new to GraphQL, so I might be doing something wrong here. Anyway, I am trying to deprecate some fields in our schema using the @deprecated(reason: "reason") directive. When I add the directive, the deprecated fields are no longer visible in GraphiQL and if I try to use the deprecated field in a query I get this error message: "Field is not defined by type . Did you mean <other field name 1> or <other field name 2>?"

I expected that the deprecated fields wouldn't be discoverable, but that they would show up with a warning that they are deprecated. However, the fields aren't even showing up for me. Here's a quick example:

Schema snippet:
```
type Query {
    securities(
        masterSecurityIds: [Int!]!,
        enhanced: Boolean = false @deprecated(reason: "some reason")
    ): [Security]
}
```

Screenshot from GraphiQL showing the "enhanced" argument isn't available:
![image](https://github.com/graphql/graphiql/assets/90796309/964aae17-8bb5-4aba-a42b-d0fce2f530f4)

Do I need to do something specific so that the deprecated fields still show up in GraphiQL?

Thanks!
Paul

**Comment by TallTed** [comment]
Note that if you don't code fence `@` entities on GitHub, those users get dragged into conversations they did not intend to join... like `@deprecated` above. Just add backticks around it, like ``` `@deprecated` ```.

**Comment by acao** [comment]
@pjungels-paraport hmmm.. I wonder, is there perhaps a bug in your server implementation then? the error message seems to indicate that the server doesn't recognize the deprecated field. If this were the case, then it would be missing in the introspection query response, or however the schema is constructed if you use schema prop instead of http introspection.

On the other hand, if you do see the deprecated field in question in the http introspection response or in the schema instance, then we might have a bug on our end!

**Comment by pjungels-paraport** [comment]
Thanks for the response @acao ! I played around with introspection in my project and have some new findings.

Running this query:

```
{
  __type(name:"Query") {
    name
    fields {
      name
      args(includeDeprecated: true) {
        name
        type {
          name
          description
        }
        deprecationReason
      }
    }
  }
}
```

returned this result:
```
{
  "data": {
    "__type": {
      "name": "Query",
      "fields": [
        {
          "name": "securities",
          "args": [
            {
              "name": "masterSecurityIds",
              "type": {
                "name": null,
                "description": null
              },
              "deprecationReason": null
            },
            {
              "name": "enhanced",
              "type": {
                "name": "Boolean",
                "description": "Built-in Boolean"
              },
              "deprecationReason": "some reason"
            }
          ]
        },
        {
          "name": "_service",
          "args": []
        }
      ]
    }
  }
}
```

So the deprecated field is discoverable through introspection. However, when I try to use the deprecated argument in a query, the GraphiQL UI gives me this message:
![image](https://github.com/graphql/graphiql/assets/90796309/2e041a2d-d849-4890-9f07-31a8f0d28a20)
but the query runs successfully (in my original post, I indicated I had gotten an error message - I'm wondering if I mistyped the argument).

The GraphiQL documentation explorer still doesn't show the deprecated argument (as in my original post).
